src/tiler/server.js

```diff
@@ -63,8 +90,44 @@ var interactivity = {
             stream_order = 5;
         }
 
-        return '(SELECT geom, stream_order FROM ' + tableName +
-          ' WHERE stream_order >= ' + stream_order + ') as q';
+        var caseSql = function(mapping, field) {
+            bounds = _.map(Object.keys(mapping), function(category) {
+                return 'WHEN ' + field + ' >= ' + mapping[category][0] + ' AND ' + field +' < ' + mapping[category][1] + ' THEN \'' + category +'\' ';
+            });
+
+            bounds[0] = 'CASE ' + bounds[0];
+            bounds.push('WHEN ' + field + ' > ' + mapping.L4[1] + ' THEN \'L5\' ');
+            bounds.push('ELSE \'NA\' ');
+            bounds.push('END');
+
+            return _.reduce(bounds, function(acc, x) { return acc + ' ' + x; });
+        }
+
+        if (tableId === 'nhd_quality_tn') {
+            return '(SELECT geom, stream_order, ' + 
```

Comment:
These three queries look identical save for some parameters, I think they could be DRY'd up so that the query is constructed only once with the relevant parameters passed in.